bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigUtil.java

```diff
@@ -63,9 +67,49 @@ public static Object normalizeType(Object value, ConfigDescriptionParameter conf
         if (configDescriptionParameter != null) {
             Normalizer normalizer = NormalizerFactory.getNormalizer(configDescriptionParameter);
             return normalizer.normalize(value);
-        } else {
-            return value instanceof Double ? BigDecimal.valueOf((Double) value) : value;
+        } else if (value == null || value instanceof Boolean || value instanceof String
+                || value instanceof BigDecimal) {
+            return value;
+        } else if (value instanceof Number) {
+            return new BigDecimal(value.toString());
```

Comment:
The constructor new BigDecimal (double ) must not be used because  constructor gets all the imprecision of the double

It was written in its doc:
" 
....
<b>Notes:</b>
     * The results of this constructor can be somewhat unpredictable.
     * One might assume that writing {@code new BigDecimal(0.1)} in
     * Java creates a {@code BigDecimal} which is exactly equal to
     * 0.1 (an unscaled value of 1, with a scale of 1), but it is
     * actually equal to
     * 0.1000000000000000055511151231257827021181583404541015625.
     * This is because 0.1 cannot be represented exactly as a
     * {@code double} (or, for that matter, as a binary fraction of
     * any finite length).  Thus, the value that is being passed
     * <i>in</i> to the constructor is not exactly equal to 0.1,
     * appearances notwithstanding.
.....
"

You can try this code:

        double c = 47.48;
        BigDecimal b = new BigDecimal(c);
        System.out.println(b);
        // This prints 47.47999999999999687361196265555918216705322265625